### PR TITLE
Updates Developer's Guide

### PR DESCRIPTION
**Pull request by nordic96**
Added Use Case (UC07)
Changed title to Typee-Developer's Guide
Added High-level architecture diagram

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -412,7 +412,14 @@ Use case resumes at step 2.
 
 //
 [discrete]
-=== Use case: (UC07) Load and save data
+=== Use case: (UC07) Save updated Appointment data
+*MSS*
+
+1. User make changes in the appointment list, or a specific appointment via CRUD.
+2. System saves the updated data to the external text file
+3. System displays the updated appointment data in the main window
```
Comment (jun-ha0) [inline docs/DeveloperGuide.adoc]: Might be too specific.
How about writing
2. System saves the updated data
3. System displays the updated appointment data

Architecture diagram LGTM 👍

Comment (lyskevin) [reply docs/DeveloperGuide.adoc]: I agree with Jun Hao on points 2 and 3 but otherwise it looks good :)

Comment (nordic96) [reply docs/DeveloperGuide.adoc]: Updated UC07 accordingly. Refer to 8d1aff4

Comment (nordic96) [reply docs/DeveloperGuide.adoc]: Merging pull request

**Pull request merged**